.github/workflows/build_and_test.yml

```diff
@@ -53,6 +53,9 @@ on:
       codecov_token:
         description: The upload token of codecov.
         required: false
+concurrency:
+  group: build-test-${{ github.workflow }}-${{ github.repository == 'apache/spark' && github.run_id || github.ref }}
+  cancel-in-progress: true
```

Comment:
It seems we can follow the [example](https://docs.github.com/en/enterprise-cloud@latest/actions/how-tos/write-workflows/choose-when-workflows-run/control-workflow-concurrency#example-only-cancel-in-progress-jobs-on-specific-branches) with
```
cancel-in-progress: ${{ github.repository != 'apache/spark'}}
```